lib/http/2/error.rb

```diff
@@ -25,7 +25,7 @@ class FlowControlError < ProtocolError; end
 
     # Raised on invalid stream processing: invalid frame type received or
     # sent, or invalid command issued.
-    class StreamError < ProtocolError; end
+    class InternalError < ProtocolError; end
```

Comment:
From Stream#stream_error
If stream_error is supposed to be a defined error, then we can add it an remove this change, I simply assumed it was not.

```ruby

    def stream_error(error = :stream_error, msg: nil)
      @error = error
      close(error) if @state != :closed   # Here, IUC, error needs to be a valid defined error

      klass = error.to_s.split('_').map(&:capitalize).join
      fail Error.const_get(klass), msg  # Here, the Exception class name needs to match the error name
    end
```